modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -77,7 +81,16 @@ object ConsensusManager {
           def getObservedOutcome: F[ConsensusOutcome[Key, Artifact]] = for {
             readyPeers <- clusterStorage.getResponsivePeers
               .map(_.filter(_.state === Ready))
-            selectedPeer <- Random[F].elementOf(readyPeers)
+            numberOfPeersToQuery <- Random[F].nextIntBounded(readyPeers.size / 4 + 1)
+            peersToQuery <- Random[F]
```

Comment:
If you rely on the majority, why you don't query all the peers. If I understood this correctly, by querying a subset of peers, you'll build up wrong majority observation.

Reply:
If the short term, querying all the peers wouldn't cause much of an issue, but it's not ideal at much higher node counts. We've made a few other tickets for adding a random sampling by weighted trust to ensure nodes are sampled with respect to overall score -- and additionally to observe the gossip stream to collect data -- so planned improvements to this would be done separately (ideally after the initial single peer join process is improved). If every node restart incurs a bulk broadcast to all nodes, it's not really great. Furthermore, starting from now this is at least a minor improvement to dealing with a random peer.

Extending this to the full list would be okay in short term however, but not critical